lms/djangoapps/program_enrollments/management/commands/link_program_enrollments.py

```diff
@@ -0,0 +1,134 @@
+""" Management command to link program enrollments and external student_keys to an LMS user """
+import logging
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand, CommandError
+from django.db import transaction
+from lms.djangoapps.program_enrollments.models import ProgramEnrollment
+from student.models import CourseEnrollmentException
+
+logger = logging.getLogger(__name__)
+User = get_user_model()
+
+INCORRECT_PARAMETER_TPL = u'incorrectly formatted argument {}, must be in form <external user key>:<lms username>'
+DUPLICATE_KEY_TPL = u'external user key {} provided multiple times'
+NO_PROGRAM_ENROLLMENT_TPL = u'No program enrollment found for program uuid={} and external student key={}'
+NO_LMS_USER_TPL = u'No user found with username {}'
+COURSE_ENROLLMENT_FAILURE_TPL = u'Failed to enroll user {} with waiting program course enrollment for course {}'
+EXISTING_USER_TPL = u'Program enrollment with external_student_key={} is already linked to {} account username={}'
+
+
+class Command(BaseCommand):
+    # pylint: disable=missing-docstring
```

Comment:
nit: maybe just write a docstring?

Reply:
...especially capture things like: what happens when you provide an external key that we don't have a program enrollment record for?  or similarly, a username for which there is no user?

Reply:
also, example usage is always good to have